### PR TITLE
KIALI-1543 - continuing list filters refactoring

### PR DESCRIPTION
**Pull request by jotak**
This PR has 2 commits:
- ab2f864 takes back the rollbacked PR #667
- 1527eed fixes found issues afterward (KIALI-1562)

The first one was already reviewed, the addition bring by the second are:
- Move more logic into StatefulFilters
- Now shared-memory filters is the "source of truth" (before i was more ambiguous between URL and shared mem)
- As a consequence, when linking with filter, 'onclick' must be set to update shared mem (it was sometimes done before, but not always)
- Some other bits simplified, hopefully not too much

- Also, from Summary Graph panel, replace the "Services list" link to Apps list, as it seems to be a forgotten step from earlier devs.

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#65](https://paste.fedoraproject.org/paste/xCsxeVosBXkkgQfKxLHsUw)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1288]](https://paste.fedoraproject.org/paste/UiwMUy6zwxToWu2XwsCdmg)
  * :heavy_check_mark: Istio-Upstream [#[172]](https://paste.fedoraproject.org/paste/vryDLmoEDPC9M3hZPmNMEg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1094, 1095]](https://paste.fedoraproject.org/paste/KLqaAMFRa5I0IUNsmKSgCg)
  * :heavy_check_mark: install-bookinfo [#[638]](https://paste.fedoraproject.org/paste/ZCGvtVzESf9smsb6LY1rKw)
  * :heavy_check_mark: Kiali-on-Openshift [#[640]](https://paste.fedoraproject.org/paste/00KfS1JRc6AMPHrxZjIGyA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[104]](https://paste.fedoraproject.org/paste/m~m9ksyDNPbNyAzeBwadQA)


KIALI_NAME=kialiuipr680-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#66](https://paste.fedoraproject.org/paste/n-YWpJS0x1j6AQi1COO~2w)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1289]](https://paste.fedoraproject.org/paste/ZziNzJzUm1P712JWmhDeoA)
  * :heavy_check_mark: Istio-Upstream [#[173]](https://paste.fedoraproject.org/paste/WS4vYpTP650~K1uJSJV4ww)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1096, 1097]](https://paste.fedoraproject.org/paste/CTjSHRH1ttI8-eQ0shA3bQ)
  * :heavy_check_mark: install-bookinfo [#[639]](https://paste.fedoraproject.org/paste/DnJMPVZ1SBJbziDzTRNIwA)
  * :heavy_check_mark: Kiali-on-Openshift [#[641]](https://paste.fedoraproject.org/paste/Ffp1-TC7KYE1njUEtbH6YQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[105]](https://paste.fedoraproject.org/paste/KjsZKGhfZZ~9xe6abP8qlA)


KIALI_NAME=kialiuipr680-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#67](https://paste.fedoraproject.org/paste/8Zv1CQ~EnXhuTOpdWR~eJA)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1290]](https://paste.fedoraproject.org/paste/Ff~fdrRtwwluv0eyvzuX2Q)
  * :heavy_check_mark: Istio-Upstream [#[174]](https://paste.fedoraproject.org/paste/U-CpAMVkwH~Yq9GqTmROeg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1098, 1099]](https://paste.fedoraproject.org/paste/mrf2cib0YAKU8g~46HJcnQ)
  * :heavy_check_mark: install-bookinfo [#[640]](https://paste.fedoraproject.org/paste/ljF8uG6muRgflfC-irap-Q)
  * :heavy_check_mark: Kiali-on-Openshift [#[642]](https://paste.fedoraproject.org/paste/GNDwOs6626olx6E9Q66Axg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[106]](https://paste.fedoraproject.org/paste/vcfht5JqsWJnKMscffURSg)


KIALI_NAME=kialiuipr680-istio-system

**Comment by xeviknal** [comment]
I successfully tested the following features locally:

* Filters, sort field, sort direction, rate interval and pagination (page, per page):
	* Are mapped to the URL.
	* when refreshing page, those params are applied to the list.
* When switching from list pages:
	* Filters that doesn't apply to the new list just disappear.
	* Only filters are brought. sorting and rate interval aren't.
* Boolean filters doesn't append (i.e.: istio sidecar, app label, version label)
* Pagination's Page gets reset when adding/removing filters. It means that is moved back to page 1.
* Breadcrumb:
	* Clicking to first item, move user to an unfiltered list.
	* Clicking to followin items, move user to a filtered list (e.g. by namespace).

File: src/components/Filters/StatefulFilters.tsx [hunk 1]
```diff
@@ -70,42 +79,24 @@ export class StatefulFilters extends React.Component<StatefulFiltersProps, State
   }
 
   componentDidUpdate(prevProps: StatefulFiltersProps, prevState: StatefulFiltersState, snapshot: any) {
-    const filtersExists = (prevProps.initialActiveFilters || []).every(prevFilter => {
```
Comment (lucasponce) [inline src/components/Filters/StatefulFilters.tsx]: Where is this logic replaced ?
I miss the trace with the whole refactoring, where is checked duplication on filters ?

Comment (jotak) [reply src/components/Filters/StatefulFilters.tsx]: If I understand correctly this code, it's more about checking if there is any change in filters (between prev props and new props) rather than anything about duplicate filters.
The logic is a bit changed and now `filtersMatchURL` plays an similar role.
I'm myself a bit scared about not missing a hidden corner case when I look at the pieces of code removed :-s So manual tests are quite important here.

I just tested that duplicated filters are correctly handled, and it's ok. In StatefulFilters.tsx it's still managed, cf function `duplicatesFilter`  which is still there.

Comment (lucasponce) [reply src/components/Filters/StatefulFilters.tsx]: My preliminar smoke test looks ok, nothing seems broken so far.
Anyway, @hhovsepy can also help to review this PR, at least to check nothing is missing.
